### PR TITLE
Add drawerLockMode prop to DrawerLayoutAndroid

### LINKED ISSUE #5270: how can i set DrawerLayoutAndroid lock mode to DrawerLayout.LOCK_MODE_LOCKED_CLOSED ?
how can i set DrawerLayoutAndroid lock mode to DrawerLayout.LOCK_MODE_LOCKED_CLOSED ?

### PR DESCRIPTION
**Pull request by andybb**
Closes #5270.

### CODE REVIEW

File: Libraries/Components/DrawerAndroid/DrawerLayoutAndroid.android.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ var INNERVIEW_REF = 'innerView';
 var DrawerLayoutValidAttributes = {
   drawerWidth: true,
   drawerPosition: true,
+  drawerLockMode: true
```
Comment (andybb) [inline Libraries/Components/DrawerAndroid/DrawerLayoutAndroid.android.js]: Is this needed?

File: Libraries/Components/DrawerAndroid/DrawerLayoutAndroid.android.js [hunk 2]
```diff
@@ -96,6 +98,14 @@ var DrawerLayoutAndroid = React.createClass({
      */
     drawerWidth: ReactPropTypes.number,
     /**
+     * Specifies the lock mode of the drawer.
```
Comment (mkonicek) [inline Libraries/Components/DrawerAndroid/DrawerLayoutAndroid.android.js]: What is a lock mode? Can you add documentation understandable for someone who doesn't know anything about the Android Drawer?

Comment (andybb) [inline Libraries/Components/DrawerAndroid/DrawerLayoutAndroid.android.js]: Better now?

File: Libraries/Components/DrawerAndroid/DrawerLayoutAndroid.android.js [hunk 3]
```diff
@@ -96,6 +98,18 @@ var DrawerLayoutAndroid = React.createClass({
...
+     * - unlocked (default), meaning that the drawer will respond (open/close) to touch gestures.
+     * - locked closed, meaning that the drawer will stay closed and not respond to gestures.
+     * - locked open, meaning that the drawer will stay opened and not respond to gestures.
+     * The drawer may still be opened and closed programmatically (`openDrawer`/`closeDrawer`).
+     */
+    drawerLockMode: ReactPropTypes.oneOf([
+      DrawerConsts.DrawerLockMode.Unlocked,
+      DrawerConsts.DrawerLockMode.LockedClosed,
+      DrawerConsts.DrawerLockMode.LockedOpen
+    ]),
```
Comment (satya164) [inline Libraries/Components/DrawerAndroid/DrawerLayoutAndroid.android.js]: @nicklockwood was suggesting not to use constants. These could be just strings, e.g. - "open", "closed", "unlocked".

**Comment by satya164** [comment]
@andybb Any upadtes?

**Comment by andybb** [comment]
@satya164 I can rewrite to strings, but what about this pull request? https://github.com/facebook/react-native/pull/5880 Nearly the same.

**Comment by satya164** [comment]
Let's merge this one after fixing the strings.

**Comment by mkonicek** [comment]
:+1:

**Comment by andybb** [comment]
@satya164 Done!

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Comment by satya164** [comment]
Awesome!

**Pull request closed without merging**